src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1532,7 +1541,7 @@ protected internal virtual TypeDefinition MarkType (TypeReference reference, Dep
 				// will call MarkType on the attribute type itself). 
 				// If for some reason we do keep the attribute type (could be because of previous reference which would cause IL2045
 				// or because of a copy assembly with a reference and so on) then we should not spam the warnings due to the type itself.
-				if (sourceLocationMember.DeclaringType != type)
+				if (sourceLocationMember != type && sourceLocationMember.DeclaringType != type)
```

Comment:
Not really questioning this - but when is the case that type A marks "itself"? (which is the case where source == type).

Reply:
Maybe this doesn't make sense - I had made the sourceLocationMember the type itself when fully marking a "pending" marked type - but it should probably just be null.

Reply:
Replaced it with a null check